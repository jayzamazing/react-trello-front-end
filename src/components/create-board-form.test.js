'use strict';
import React from 'react';
import component, {CreateBoardForm} from './create-board-form';
import {shallow, mount} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('create-board-form', () => {
  //ensure that it renders without issues
  it('should render', () => {
    const handleSubmit = jest.fn();
    shallow(<CreateBoardForm handleSubmit={handleSubmit}/>);
  });
  //take snapshot for future changes
  it('snapshot', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(<CreateBoardForm handleSubmit={handleSubmit}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //click on create button and check dispatches
  it('add board click', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(<CreateBoardForm handleSubmit={handleSubmit}/>);
    wrapper.find('[name="boardTitle"]').at(0).simulate('change', { target: { value: 'testing' } });
    wrapper.find('.create-board-input-area').simulate('submit');
    expect(handleSubmit).toHaveBeenCalled();
  });
  //click on create button and check dispatches
  it('close button', () => {
    const handleSubmit = jest.fn();
    const closeModal = jest.fn();
    const wrapper = shallow(<CreateBoardForm closeModal={closeModal} handleSubmit={handleSubmit}/>);
    wrapper.find('[aria-label="close button"]').simulate('click');
    expect(closeModal).toHaveBeenCalled();
  });
});
