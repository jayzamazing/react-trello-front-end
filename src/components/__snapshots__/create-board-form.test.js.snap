// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-board-form snapshot 1`] = `
<form
  className="create-board-input-area"
  name="createBoardForm"
>
  <Field
    component={[Function]}
    inputClass="create-board-input"
    labelclass="remove"
    name="boardTitle"
    placeholder="Add board title"
    type="text"
    validate={
      Array [
        [Function],
        [Function],
        [Function],
      ]
    }
  />
  <button
    aria-label="close button"
    className="btn btn-default close-modal-btn"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-remove"
    />
  </button>
  <button
    className="create-board-btn btn"
    name="createBoardBtn"
    type="submit"
  >
    Create Board
  </button>
</form>
`;
